pkg/destroy/azure/azure.go

```diff
@@ -578,14 +578,18 @@ func extractODataError(err error) error {
 }
 
 func deleteApplicationRegistrations(ctx context.Context, graphClient *msgraphsdk.GraphServiceClient, logger logrus.FieldLogger, infraID string) error {
-	var errorList []error
-
 	tag := fmt.Sprintf("kubernetes.io_cluster.%s=owned", infraID)
 	servicePrincipals, err := getServicePrincipalsByTag(ctx, graphClient, tag, infraID)
 	if err != nil {
 		return errors.Wrap(err, "failed to gather list of Service Principals by tag")
 	}
+	// msgraphsdk can return a `nil` response even if no errors occurred
```

Comment:
This logic can be moved within https://github.com/openshift/installer/blob/master/pkg/destroy/azure/azure.go#L621. Also, looking at getServicePrincipalsByTag(), it appears that this method can set resp to nil even if the original API did not return a nil.

Reply:
I preferred to do it here so I don't have to change the `getServicePrincipalsByTag` to receive a `logger` just so I can log a message in case that happens. I can change it if you think there is an advantage in doing that.

Reply:
> Also, looking at getServicePrincipalsByTag(), it appears that this method can set resp to nil even if the original API did not return a nil.

That's OK because in that case `err != nil` and the error condition will (and should) be evaluated first.